python/src/functions/generate_presigned_url_and_send_email.py

```diff
@@ -0,0 +1,133 @@
+import json
+import os
+import tempfile
+from datetime import datetime
+from typing import IO, Dict, List, Set, Union
+
+import boto3
+import chevron
+import html2text
+import structlog
+from aws_lambda_typing.context import Context
+from mypy_boto3_s3.client import S3Client
+from openpyxl import Workbook, load_workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from pydantic import BaseModel
+
+from src.lib.constants import OUTPUT_TEMPLATE_FILENAME_BY_PROJECT
+from src.lib.logging import get_logger, reset_contextvars
+from src.lib.s3_helper import get_presigned_url
+from src.lib.treasury_generation_common import (
+    OrganizationObj,
+    UserObj,
+)
+from src.lib.email import send_email
+
+
+treasury_email_html = """
+Hello,
+
+Your treasury report can be downloaded <a href={url}>here</a>.
+"""
+
+treasury_email_text = """
+Hello,
+
+Your treasury report can be downloaded here: {url}.
+"""
+
+class ProjectLambdaPayload(BaseModel):
+    organization: OrganizationObj
+    user: UserObj
+
+
+@reset_contextvars
+def handle(event: ProjectLambdaPayload, context: Context):
```

Comment:
@vshia just a note that the naming of the payload and some of the logs on this function will need to be updated. I assume this is not yet changed due to the draft nature of the PR.